editor/src/clj/editor/gui.clj

```diff
@@ -843,83 +843,150 @@
-      (g/update-property node-id :layout->prop->override eutil/dissoc-in [current-layout prop-kw]))))
-
-(defn layout-property-edit-type-clear-impl [changes-fn node-id prop-kw]
-  (let [[current-layout cleared-prop-kws]
-        (g/with-auto-evaluation-context evaluation-context
-          (let [trivial-gui-scene-info (g/valid-node-value node-id :trivial-gui-scene-info evaluation-context)
-                current-layout (:current-layout trivial-gui-scene-info)
-                cleared-prop-kws (keys (changes-fn evaluation-context node-id prop-kw nil nil))]
-            (pair current-layout cleared-prop-kws)))]
-    (when (coll/not-empty cleared-prop-kws)
-      (if (str/blank? current-layout)
-        (coll/mapcat
-          #(g/clear-property node-id %)
-          cleared-prop-kws)
-        (g/update-property
-          node-id :layout->prop->override
-          update current-layout
-          (fn [prop->override]
-            (coll/not-empty
-              (apply dissoc prop->override cleared-prop-kws))))))))
+    (layout-property-clears-tx-data current-layout node-id [prop-kw])))
+
+(defn- basic-layout-property-update-in-current-layout
+  [evaluation-context node-id prop-kw update-fn & args]
+  (let [old-value (g/node-value node-id prop-kw evaluation-context)
+        new-value (apply update-fn old-value args)
+        trivial-gui-scene-info (g/valid-node-value node-id :trivial-gui-scene-info evaluation-context)
+        current-layout (:current-layout trivial-gui-scene-info)]
+    (layout-property-changes-tx-data current-layout node-id {prop-kw new-value})))
+
+(defn- layout-property-set-in-specific-layout
+  "Uses the changes-fn associated with each property to create a series of
+  transaction steps that will set or clear properties on a GuiNode in a
+  specific layout. Nil is not a valid value for a layout property, but nil
+  values can be used to clear overrides on specific properties."
+  [evaluation-context layout-name node-id & prop-kws-and-values]
+  {:pre [(string? layout-name)]}
+  (let [kv-count (count prop-kws-and-values)]
+    (when (pos? kv-count)
+      (assert (even? kv-count))
+      (let [basis (:basis evaluation-context)
+            node (g/node-by-id basis node-id)
+            prop->value-delay (delay (prop->value-for-specific-layout node-id layout-name evaluation-context))
+
+            changes
+            (coll/transfer prop-kws-and-values {}
+              (partition-all 2)
+              (mapcat (fn [[prop-kw new-value]]
+                        (if-let [changes-fn (:changes-fn (g/node-property-dynamic node prop-kw :edit-type evaluation-context))]
+                          (let [old-value (get @prop->value-delay prop-kw)]
+                            (changes-fn evaluation-context node-id prop-kw old-value new-value))
+                          {prop-kw new-value}))))]
+
+        (coll/not-empty
+          (layout-property-changes-tx-data layout-name node-id changes))))))
```

Comment:
This function could be used to set property values in a specific layout, similar to how an edit from the Properties View would work.